File: csharp/DVSClient/DVSClient/address/layout/AppliesTo.cs
using DVSClient.Common;
using DVSClient.Server.Address.Layout;

namespace DVSClient.Address.Layout
{
    public class AppliesTo
    {
        public IEnumerable<Dataset> Datasets { get; set; }
        public string? Language { get; set; }
        public string? Script { get; set; }

        public AppliesTo(Dataset dataset)
            : this(new List<Dataset> { dataset }, null, null)
        {
        }

        public AppliesTo(List<Dataset> datasets)
            : this(datasets, null, null)
        {
        }

        public AppliesTo(RestApiAddressLayoutAppliesTo appliesTo)
        {
            Datasets = appliesTo.Datasets != null ? appliesTo.Datasets.Select(Dataset.FromCode).ToList() : new List<Dataset>();
            Language = appliesTo.Language ?? string.Empty;
            Script = appliesTo.Script ?? string.Empty;
        }

        public AppliesTo(IEnumerable<Dataset> datasets, string? language, string? script)
        {
            Datasets = datasets;
            Language = language;
            Script = script;
        }

        public Country? GetCountry()
        {
            return Datasets != null && Datasets.Any() ? Datasets.ElementAt(0).Country : null;
        }
    }
}

File: typescript/src/address/layout/appliesTo.ts
import { RestApiAddressLayoutAppliesTo } from '../../server/address/layout/restApiAddressLayoutAppliesTo';
import { Dataset, Datasets } from '../dataset';

export type AppliesTo = {
    datasets: Dataset[];
    language?: string;
    script?: string;
};

export function restApiResponseToAppliesTo(apiItem: RestApiAddressLayoutAppliesTo): AppliesTo {
    return {
        datasets: apiItem.datasets ? apiItem.datasets.map(p => Datasets.fromCode(p)) : [],
        language: apiItem.language??"",
        script: apiItem.script??""
    }
}